tools/state-parts-dump/src/rpc_requests.rs

```diff
@@ -0,0 +1,214 @@
+use serde::Deserialize;
+use std::error::Error;
+
+#[derive(Deserialize)]
+struct ValidatorsResponse {
+    result: ValidatorsInfo,
+}
+
+#[derive(Deserialize)]
+struct BlockResponse {
+    result: BlockInfo,
+}
+
+#[derive(Deserialize)]
+struct StatusResponse {
+    result: StatusInfo,
+}
+
+#[derive(Deserialize)]
+struct StatusInfo {
+    sync_info: SyncInfo
+}
+
+#[derive(Deserialize)]
+struct SyncInfo {
+    latest_block_hash: String
+}
+
+#[derive(Deserialize)]
+struct ValidatorsInfo {
+    epoch_start_height: u64,
+    epoch_height: u64
+}
+
+#[derive(Deserialize)]
+struct BlockInfo {
+    header: BlockInfoHeader,
+    chunks: Vec<ChunkInfo>
+}
+
+#[derive(Deserialize)]
+struct BlockInfoHeader {
+    hash: String,
+    prev_hash: String,
+    next_epoch_id: String,
+    epoch_id: String
+}
+
+#[derive(Deserialize)]
+struct ChunkInfo {
+    prev_state_root: String,
+}
+
+pub struct RpcClient {
+    client: reqwest::blocking::Client,
+    rpc_endpoint: String,
+}
+
+impl RpcClient {
+    // network can only be "mainnet" or "testnet"
+    pub fn new(network: &str) -> Self {
+        Self { client: reqwest::blocking::Client::new(), rpc_endpoint: format!("https://rpc.{}.near.org", network) }
+    }
+
+    pub fn get_block_hash(&self, block_height: u64) -> Result<String, Box<dyn Error>> {
+        let block_response: BlockResponse = self
+            .client
+            .post(&self.rpc_endpoint)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": "dontcare",
+                "method": "block",
+                "params": {
+                    "block_id": block_height
+                }
+            }))
+            .send()?
+            .json()?;
+        let block_hash = block_response.result.header.hash;
+
+        Ok(block_hash)
+    }
+
+    // epoch_id here is the epoch_id you want state part for
+    pub fn get_current_epoch_first_block_hash(&self, epoch_id: &String) -> Result<String, Box<dyn Error>> {
+        let epoch_response: ValidatorsResponse = self
+            .client
+            .post(&self.rpc_endpoint)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": "dontcare",
+                "method": "validators",
+                "params": {
+                    "epoch_id": epoch_id
+                }
+            }))
+            .send()?
+            .json()?;
+        let epoch_start_height = epoch_response.result.epoch_start_height;
```

Comment:
nice, I always forget about this RPC as its epoch_id parameter is not documented https://docs.near.org/api/rpc/network#validation-status